### ISSUE TITLE
Job cannot br re-enqueue

### ISSUE THREAD

**Issue by michaelsogos** [sergeyzwezdin/Hangfire.Mongo#83]
Each time we try to re-enqueue the job from dashboard an MongoDriver exception will be encountered
```

MongoDB.Driver.MongoWriteException
A write operation resulted in an error. E11000 duplicate key error collection: Hangfire.hangfire.jobQueue index: _id_ dup key: { : "5985b692fc98160e48f9cefb" } 
MongoDB.Driver.MongoWriteException: A write operation resulted in an error.
  E11000 duplicate key error collection: Hangfire.hangfire.jobQueue index: _id_ dup key: { : "5985b692fc98160e48f9cefb" } ---> MongoDB.Driver.MongoBulkWriteException`1[Hangfire.Mongo.Dto.JobQueueDto]: A bulk write operation resulted in one or more errors.
  E11000 duplicate key error collection: Hangfire.hangfire.jobQueue index: _id_ dup key: { : "5985b692fc98160e48f9cefb" }
   at MongoDB.Driver.MongoCollectionImpl`1.BulkWrite(IEnumerable`1 requests, BulkWriteOptions options, CancellationToken cancellationToken)
   at MongoDB.Driver.MongoCollectionBase`1.InsertOne(TDocument document, InsertOneOptions options, CancellationToken cancellationToken)
   --- End of inner exception stack trace ---
   at MongoDB.Driver.MongoCollectionBase`1.InsertOne(TDocument document, InsertOneOptions options, CancellationToken cancellationToken)
   at Hangfire.Mongo.PersistentJobQueue.Mongo.MongoJobQueue.Enqueue(String queue, String jobId)
   at Hangfire.Mongo.MongoWriteOnlyTransaction.Commit()
   at Hangfire.States.BackgroundJobStateChanger.ChangeState(StateChangeContext context, BackgroundJob backgroundJob, IState toState, String oldStateName)
```

Can you help with this issue?

**Comment by MarLoe** [comment]
Can you please state version info for Hangfire, Hangfire.Mongo and Mongo.Driver. 

I will then look into it and see if I can reproduce.

**Comment by michaelsogos** [comment]
From NuGet:
Hangfire.Core 1.6.14
Hangfire.AspNetCore 1.6.14
Hangfire.Mongo 0.4.1
Microsoft.AspNetCore.All 2.0.0-preview2

Mongo driver is a your dependency, and seems to be MongoDB.Driver 2.3.0

While MongoDB Server is 3.4.6 CE

Is it enought? :)

**Comment by gottscj** [comment]
@MarLoe,

It seems like the job has been requeued before removed from the job queue? Im not sure how this can happen.
we could maybe do an upsert, to be on the safe side, but that would override the other job. IFetchedJob has a Requeue()

**Comment by MarLoe** [comment]
@michaelsogos, I have just been able to reproduce. I will look further into this asap.

**Comment by michaelsogos** [comment]
There are some strange behaviour with our Hangfire implementation because from documentation seems that abnormally stopped jobs should retry as soon as possible, but we never see it happens, that's why we enter in dashboard and force to retry, but we didn't push button two or more times, just one :)

However we cannot figure out what is the issue with hangfire about automatic retry.
To Repro:
Implement hangfire (thanks to the nuget libraries above listed) as described in quickstart (simply without Sql Server), than starts the application with a breakpoint before enqueueing a job (in fire-and-forget style) and a breakpoint after enqueued the job (that we never reach in our regresion test); if you have a breakpoint also into the method called by the job then the debugger should stop also there; at this point stop VS2017 debugging then start again the application as it is. Going to the dashboard you will see that the job as been aborted (even if it is under processing), and in this case if we push the button to re-enqueue the job than it encounter the exception on top described.

We are also investigating to get the reason why we don't see the job to automatically retry itself, our developers guess that maybe it happens because we are using mongo driver instead of sql server as suggested in documentation.

**Comment by MarLoe** [comment]
This is fixed by #88. It will be a part of the next release.
